### PR TITLE
Add controller unit tests

### PR DESCRIPTION
**Pull request by iamniting**
Signed-off-by: Nitin Goyal <nigoyal@redhat.com>

### CODE REVIEW

File: controllers/subscriptions_test.go [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+		existingSubscriptions := &operatorv1alpha1.SubscriptionList{}
+		err = fakeReconciler.Client.List(context.TODO(), existingSubscriptions)
+		assert.NoError(t, err)
+
+		if !tc.expectedSubscription {
+			assert.Equal(t, 0, len(existingSubscriptions.Items))
+		} else {
+			assert.Equal(t, 1, len(existingSubscriptions.Items))
+			assert.Equal(t, "ibm-operator", existingSubscriptions.Items[0].Spec.Package)
+			assert.Equal(t, "alpha", existingSubscriptions.Items[0].Spec.Channel)
```
Comment (jarrpa) [inline controllers/subscriptions_test.go:81]: This is fine for now, but I would like to see this be a little less hardcoded. Add a comment like `//TODO: Parameterize this`.

**Review by jarrpa — CHANGES_REQUESTED** [review]
Overall looks good. I would request that you add labels to the test cases so we can get descriptive messages of what each case is actually testing. You can see an example here: https://github.com/openshift/ocs-operator/blob/master/controllers/persistentvolume/persistentvolume_controller_test.go (note specifically the `Logf()` in the test loop.

File: controllers/subscriptions_test.go (hunk 1 above)
Comment (iamniting) [reply controllers/subscriptions_test.go]: Done

**Comment by iamniting** [comment]
> Overall looks good. I would request that you add labels to the test cases so we can get descriptive messages of what each case is actually testing. You can see an example here: https://github.com/openshift/ocs-operator/blob/master/controllers/persistentvolume/persistentvolume_controller_test.go (note specifically the `Logf()` in the test loop.

Done

**Review by jarrpa — APPROVED** [review]
LGTM

**Pull request merged**